onnxruntime/contrib_ops/cuda/bert/attention_transpose.cu

```diff
@@ -115,6 +167,35 @@ __global__ void TransposeQKV(const int H, const T* input, T* output) {
   }
 }
 
+template <typename T>
+__global__ void TransposeQKVLarge(const int H, const T* input, T* output) {
```

Comment:
From implementation, it seems that TransposeQKVLarge could have similar performance as TransposeQKV. (For normal case, it has an additional add and check while condition). 

Could you do a performance test? If performance is similar, we could replace TransposeQKV by TransposeQKVLarge implementation. Similar for other two kernels.